front/src/modules/pathfinding/utils.ts

```diff
@@ -0,0 +1,124 @@
+  operationalPoints: NonNullable<Required<PathProperties['operational_points']>>,
+  geometry: GeoJsonLineString,
+  pathLength: number
+): SuggestedOP[] =>
+  operationalPoints.map((op) => ({
+    opId: op.id,
+    name: op.extensions?.identifier?.name,
+    uic: op.extensions?.identifier?.uic,
+    ch: op.extensions?.sncf?.ch,
+    chLongLabel: op.extensions?.sncf?.ch_long_label,
+    chShortLabel: op.extensions?.sncf?.ch_short_label,
+    ci: op.extensions?.sncf?.ci,
+    trigram: op.extensions?.sncf?.trigram,
+    offsetOnTrack: op.part.position,
+    track: op.part.track,
+    positionOnPath: op.position,
+    coordinates: getPointCoordinates(geometry, pathLength, op.position),
+  }));
+
+export const getPathfindingQuery = ({
+  infraId,
+  rollingStock,
+  origin,
+  destination,
+  pathSteps,
+}: {
+  infraId?: number;
+  rollingStock?: RollingStockWithLiveries;
+  origin: PathStep | null;
+  destination: PathStep | null;
+  pathSteps: (PathStep | null)[];
+}): PostV2InfraByInfraIdPathfindingBlocksApiArg | null => {
+  if (infraId && rollingStock && origin && destination) {
+    // Only origin and destination can be null so we can compact
+    const pathItems: PathfindingInputV2['path_items'] = compact(pathSteps).map((step) => {
+      if ('uic' in step) {
+        return { uic: step.uic };
+      }
+      if ('track' in step) {
+        return {
+          track: step.track,
+          offset: step.offset,
+        };
+      }
+      if ('operational_point' in step) {
+        return {
+          operational_point: step.operational_point,
+        };
+      }
+      return {
+        trigram: step.trigram,
+      };
+    });
+
+    return {
+      infraId,
+      pathfindingInputV2: {
+        path_items: pathItems,
+        rolling_stock_is_thermal: isThermal(rollingStock.effort_curves.modes),
+        rolling_stock_loading_gauge: rollingStock.loading_gauge,
+        rolling_stock_supported_electrifications: getSupportedElectrification(
+          rollingStock.effort_curves.modes
+        ),
+        rolling_stock_supported_signaling_systems: rollingStock.supported_signaling_systems,
+      },
+    };
+  }
+  return null;
+};
+
+export const insertViasInOPs = (ops: SuggestedOP[], pathSteps: PathStep[]): SuggestedOP[] => {
+  let updatedOPs = [...ops];
+  const vias = pathSteps.slice(1, -1);
+  if (vias.length > 0) {
+    vias.forEach((step) => {
+      // We check only for vias added by map click
+      if ('track' in step) {
+        const formattedStep: SuggestedOP = {
+          opId: step.id,
+          positionOnPath: step.positionOnPath!,
+          offsetOnTrack: step.offset,
+          track: step.track,
+          coordinates: step.coordinates,
+        };
+        // If it hasn't an uic, the step has been added by map click,
+        // we know we have its position on path so we can insert it
+        // at the good index in the existing operational points
+        const index = compact(updatedOPs).findIndex(
```

Comment:
I think compact is not needed here
```suggestion
        const index = updatedOPs.findIndex(
```

Reply:
Done